src/Sylius/Bundle/ArchetypeBundle/spec/Form/EventListener/ParentArchetypeListenerSpec.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Bundle\ArchetypeBundle\Form\EventListener;
+
+use PhpSpec\Exception\Example\FailureException;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Symfony\Component\Form\FormBuilder;
+
+/**
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ */
+class ParentArchetypeListenerSpec extends ObjectBehavior
+{
+    function let()
+    {
+        $this->beConstructedWith('product');
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Bundle\ArchetypeBundle\Form\EventListener\ParentArchetypeListener');
+    }
+
+    function it_implements_event_subscriber_interface()
+    {
+        $this->shouldImplement('Symfony\Component\EventDispatcher\EventSubscriberInterface');
+    }
+
+    function it_gets_subscribed_events()
+    {
+        $this->getSubscribedEvents()->shouldBeArray();
```

Comment:
`$this::getSubscribedEvents()->shouldReturn(array(FormEvents::PRE_SET_DATA => 'preSetData'));` will be enough.